vortex-array/src/accessor.rs

```diff
@@ -1,16 +1,14 @@
 // SPDX-License-Identifier: Apache-2.0
 // SPDX-FileCopyrightText: Copyright the Vortex contributors
 
-use vortex_error::VortexResult;
-
 /// Trait for arrays that support iterative access to their elements.
 pub trait ArrayAccessor<Item: ?Sized> {
     /// Iterate over each element of the array, in-order.
     ///
     /// The function `f` will be passed an [`Iterator`], it can call [`next`][Iterator::next] on the
     /// iterator [`len`][crate::Array::len] times. Iterator elements are `Option` types,
     /// regardless of the nullability of the underlying array data.
-    fn with_iterator<F, R>(&self, f: F) -> VortexResult<R>
+    fn with_iterator<F, R>(&self, f: F) -> R
```

Comment:
Note that the inner type `R` can _still_ be `VortexResult`, this outer result wrapping was from the time when Array Ops used to be fallible. But all arrays are checked at the boundaries and in practice none of the implementations ever returned Err, they just did dumb `Ok(...)` wrapping